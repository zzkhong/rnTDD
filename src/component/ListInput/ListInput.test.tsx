import React from 'react';
import {act, create} from 'react-test-renderer';
import {ListInput} from '.';

const tree = create(<ListInput />);

// Snapshot Test (UI Rendering Matching Test)
// If updates are expected (such as background color change)
// Just run `yarn test -u` to update the snapshots
test('#1 - ListInput Snapshot', () => {
  expect(tree).toMatchSnapshot();
});

// State Change
test('#2 - ListInput state input', () => {
  // Grab TextInput and trigger onChangeText
  const textInput = tree.root.findByProps({testID: 'listInput_textInput'})
    .props;
  act(() => textInput.onChangeText('New Input'));

  // Expect Text to have same value as input
  const text = tree.root.findByProps({testID: 'listInput_text'}).props;
  expect(text.children).toEqual('New Input');
  expect(text.children).not.toEqual('');
});
