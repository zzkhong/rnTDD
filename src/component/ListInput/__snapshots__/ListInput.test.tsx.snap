// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#1 - ListInput Snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFF",
    }
  }
>
  <Text
    testID="listInput_text"
  >
    
  </Text>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    rejectResponderTermination={true}
    testID="listInput_textInput"
    underlineColorAndroid="transparent"
    value=""
  />
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  />
</View>
`;
